woocommerce/google-listings-and-ads
# Update Google Content API to 2.13 and Google Ads API to V12

### Changes proposed in this Pull Request:

**Project**: pcTzPl-1kF-p2

This PR updates the Google Ads library from V11 > V12 as well as Google Content API (Merchant Centre) to 2.13.

- Update the Google Client services
- Update the Google Ads API from V11 to V12
- Change all imports from V11 > V12
- Cleanup older versions of the Ads library V9 - V11

### Updated packages
```
composer update php
composer update googleads/google-ads-php --with-all-dependencies

- Upgrading firebase/php-jwt (v6.3.0 => v6.3.2
- Upgrading google/apiclient (v2.12.6 => v2.13.0)
- Upgrading google/auth (v1.21.1 => v1.25.0)
- Upgrading google/common-protos (v2.1.0 => v3.2.0)
- Upgrading google/gax (v1.13.0 => v1.19.0)
- Upgrading google/protobuf (v3.21.2 => v3.21.12)
- Upgrading googleads/google-ads-php (v16.0.0 => v17.1.0)
- Upgrading guzzlehttp/guzzle (v7.4.5 => 7.5.0)
- Upgrading guzzlehttp/promises (1.5.1 => 1.5.1)
- Upgrading guzzlehttp/psr7 (2.4.0 => 2.4.3)
- Upgrading monolog/monolog (2.7.0 => 2.9.1)
- Upgrading phpseclib/phpseclib (3.0.14 => 3.0.18)
- Upgrading symfony/polyfill-intl-normalizer (v1.26.0 => v1.27.0)
- Upgrading symfony/polyfill-mbstring (v1.26.0 => v1.27.0)
- Upgrading google/longrunning (v0.2.3)

composer update google/apiclient --with-all-dependencies

- Upgrading google/apiclient-services (v0.258 => v0.286)
```

### Detailed test instructions:
1. Checkout this PR, remove the vendors folder and run `composer install` 
2. Run the unit tests locally to confirm they all pass without errors or warnings
3. Test the plugin and the Connection Test page and confirm we can send requests without any fatals or errors (need to connect all accounts)
4. Send both regular requests to Merchant Center and Ads as well as proxied requests in the Connection Test page.
5. In the Settings page disconnect any previously connected Ads Account
6. Onboard the same Ads account again
7. Confirm that all the steps complete without issues
8. Check the option `gla_ads_conversion_action` in the `wp_options DB table, and make sure it contains something like:
```
a:5:{
    s:2:"id";i:987654321;
    s:4:"name";s:46:"[a12b] Google Listings and Ads purchase action";
    s:6:"status";s:7:"ENABLED";
    s:13:"conversion_id";s:12:"AW-12345678";
    s:16:"conversion_label";s:20:"abc_DEF";
}
```
9. View the list of campaigns on the Dashboard page including some report data to confirm that there are no errors when using the update library


### Additional details:

A follow up PR to update the WCS to use API V12 will be created soon.

### Changelog entry
* Update - Google Ads library to API V12.
* Update - Google Content library to API 2.13.
